### PR TITLE
Add push command to deprecate apply

### LINKED ISSUE #923: Add an alias to for `fidesctl apply`
### Is your feature request related to a specific problem?

Now that we have `fidesctl pull`, it seems like `fidesctl apply` should actually be `fidesctl push`

### Describe the solution you'd like

To prevent any breakage, we should add `fidesctl push` as an identical command and also set a `DeprecationWarning` whenever `apply` gets run to warn users for future versions.

### PR DESCRIPTION
**Pull request by earmenda**
Closes #923 

### Code Changes

* [x] Add a `push` command which calls existing `apply` call
* [x] Added deprecated flag to `apply` command
* [x] Added integration tests for `push` which just follow same structure as apply. 

### Steps to Confirm

* [x] Manual Testing
* [x] Integration tests test new command and old still

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

I looked into using something fancy to solve this but it doesn't really seem worth it. This [library](https://github.com/click-contrib/click-aliases) provides the ability to add an alias to a command but the downside is that there is no option to deprecate the command. 

Open to trying a more complex change but this feels like a good bridge to eventually remove the `apply` command. 

#### Apply command output
```
fides> fides apply
DeprecationWarning: The command 'apply' is deprecated.
Use the 'push' command instead.
Loading resource manifests from: .fides/
Taxonomy successfully created.
----------
Processing system resource(s)...
PUSHED 1 system resource(s).
----------
Processing policy resource(s)...
PUSHED 2 policy resource(s).
----------
Processing dataset resource(s)...
PUSHED 1 dataset resource(s).
```
#### Push command output
```
fides> fides push 
Loading resource manifests from: .fides/
Taxonomy successfully created.
----------
Processing system resource(s)...
PUSHED 1 system resource(s).
----------
Processing dataset resource(s)...
PUSHED 1 dataset resource(s).
----------
Processing policy resource(s)...
PUSHED 2 policy resource(s).
```

### CODE REVIEW

**Comment by earmenda** [comment]
I tried doing this in a way where the `push` function just calls the cli `apply` function but it doesn't work like i expected. For now just calling the same code paths by copying what the `apply` command does. 

Still need to update documentation. We could also rename the `apply.py` file and any references to `apply` to push internally. Thoughts?

**Review by SteveDMurphy — APPROVED** [review]
I like this as an efficient way to start breaking down the `apply` command, nice work @earmenda !! 

Would it make sense to try and give a version estimate of when it will be fully pulled (i.e. `1.9`)? I can imagine there are a lot of docs updates as well so am assuming those might go in a follow on issue 👍🏽

File: src/fidesctl/cli/commands/core.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 from fidesctl.ctl.core.utils import git_is_dirty
 
 
-@click.command()
+@click.command(deprecated=True)
```
Comment (ThomasLaPiana) [inline src/fidesctl/cli/commands/core.py:21]: hey that's pretty neat 💯

File: src/fidesctl/cli/commands/core.py [hunk 2]
```diff
@@ -29,6 +29,33 @@
...
+@click.pass_context
+@dry_flag
+@click.option(
+    "--diff",
+    is_flag=True,
+    help="Include any changes between server and local resources in the command output",
+)
+@manifests_dir_argument
+@with_analytics
+def push(ctx: click.Context, dry: bool, diff: bool, manifests_dir: str) -> None:
```
Comment (ThomasLaPiana) [inline src/fidesctl/cli/commands/core.py:58]: is it possible to duplicate the command without a copy/pasta? I haven't tried it but I was hoping there'd be a clever way to do that, if not its fine, this command doesn't change much!

**Review by ThomasLaPiana — CHANGES_REQUESTED** [review]
As this is a deprecation, lets go ahead and make the docs updates here too

**Comment by ThomasLaPiana** [comment]
@conceptualshark would you be able to help us update the docs here to replace `fidesctl apply` with `fidesctl push` where needed? 🙏

File: src/fidesctl/cli/commands/core.py (hunk 2 above)
Comment (ThomasLaPiana) [reply src/fidesctl/cli/commands/core.py:58]: after looking into some of the solutions you mentioned, it's not worth the eng time to figure this out before the deprecation, I'm good with how it is 👍

**Comment by ThomasLaPiana** [comment]
@earmenda as an additional point of consistency, let's also update the logging to say `PUSHED 1 <resource>...` as opposed to `APPLIED 1 <resource>...`

**Comment by conceptualshark** [comment]
👍 think that's all the `apply` uses updated, should be good to go!

**Comment by ThomasLaPiana** [comment]
@earmenda I'm making a few more cleanup changes here and then i'll merge

**Comment by earmenda** [comment]
thanks @ThomasLaPiana

**Review by ThomasLaPiana — APPROVED** [review]

**Pull request merged**